### ISSUE TITLE
[help-wanted] TypeScript/React get component props

### ISSUE THREAD

**Issue by sanchobouillant** [brigade/react-waypoint#268]
Hi,

Is there a way to get my component's props in the function loadMoreItems ? 

    private _loadMoreItems() {
        console.log(this);

        //Get the datas from xhr, and save data to Redux
    };

**Comment by MatthewHerbst** [comment]
Hi @sanchobouillant. Could you expand on what you are trying to do please? When would you not already have access to your component's props (presumably a parent/nth-grandparent of the Waypoint)?

**Comment by sanchobouillant** [comment]
The console.log(this); return the Waypoint's object.

I'm new with React, maybe am I looking in the wrong place ? 

I want to access to my component that render Waypoint and access to my props.

**Comment by MatthewHerbst** [comment]
Right, but what I'm trying to understand is why you want to access them _inside_ of the Waypoint? You already have access to them in the render method where you render the Waypoint. Ex:

```
class MyComponent extends React.PureComponent {
  handleLoadMore() {
    // Do something, such as load more items from a paginated list
  }

  render() {
    console.log(this.props); // This will give you all the props from the component rendering the Waypoint

    return <div><Waypoint onEnter={this.handleLoadMore} /><div>;
  }
}
```
